go-controller/pkg/ovn/master.go

```diff
@@ -471,6 +483,31 @@ func (oc *Controller) SetupMaster(masterNodeName string) error {
 	return nil
 }
 
+func addNodeLogicalSwitchPort(logicalSwitch, portName, portType, addresses, options string) (string, error) {
+	stdout, stderr, err := util.RunOVNNbctl("--", "--may-exist", "lsp-add", logicalSwitch, portName,
```

Comment:
Should use go-ovn bindings since we have API support for that now. Look for similar code in `addLogicalPort` in `pods.go`

Reply:
Thanks for the review!  Yes, ideally we should use go-ovn bindings. The problem however is that we need to retrieve the UUID for [logical switch ports connected to the cluster router](https://github.com/ovn-org/ovn-kubernetes/pull/1705/files#diff-1f90f4648dc7d68c42854549c4257ddcfc9bd9a5b6fc480c74e3a23989914bc3R753).  go-ovn doesn't allow users to explicitly create such ports because it doesn't expose any API to set the port type to "router", e.g., `LSPSetType`.  Instead users should create the router ports with `LinkSwitchToRouter()`.

That means we have three options:
1. keep the current approach of using `ovn-nbctl` to create the node switches/routers, mgmt port, switch ports connected to the routers. This is what this PR does.
2. refactor most of master.go to use go-ovn bindings and especially `LinkSwitchToRouter()`.
3. add support for `LSPSetType` in the go-ovn bindings.

I'd personally go for option 1 for now and choose between 2 and 3 later as they're out of the scope of multicast support. What do you think?

Reply:
That's fine. Let's do option 1 for now.